src/parser/content-evaluator-module.ts

```diff
@@ -106,7 +123,8 @@ export class ContentEvaluatorModule implements Module {
       .map((comment) => comment)
       .join(
         "\n"
-      )}\n\`\`\`\n\n\nTo what degree are each of the comments in the conversation relevant and valuable to further defining the issue specification? Please reply with ONLY an array of float numbers between 0 and 1, corresponding to each comment in the order they appear. Each float should represent the degree of relevance and added value of the comment to the issue. The total length of the array in your response should equal exactly ${comments.length
-      } elements.`;
+      )}\n\`\`\`\n\n\nTo what degree are each of the comments in the conversation relevant and valuable to further defining the issue specification? Please reply with ONLY an array of float numbers between 0 and 1, corresponding to each comment in the order they appear. Each float should represent the degree of relevance and added value of the comment to the issue. The total length of the array in your response should equal exactly ${
```

Comment:
I think there's an official way to tell the API to return a JSON response with some type of property in the API call. 

This feature did not exist when the original code was authored

Reply:
Reference docs https://community.openai.com/t/how-do-i-use-the-new-json-mode/475890

Reply:
Ok, I was wondering what the benefit of switching to JSON would be here. I have looked into it. Previously it was sending us an array of the relevance values. To get a JSON, I have to specify the option you are pointing to. But it also requires changing the prompt to explicitly ask for a JSON response, where I will also describe the structure of JSON. 

How can this benefit us? 
I can send both the github comment id, and comment content, and get a response back with the comment id and its relevance. In this way, the chances of missing a value will be almost none.

This requires changing the prompt and how we are packing the comments & specifications for the prompt. 
Let me know if you have some other benefit in mind that you want to target.

Reply:
Here is a quickly written sample prompt for jSON. I will improve it. 
![image](https://github.com/user-attachments/assets/25c0ab8a-6b2b-4a9d-9016-c741c1ff179b)

Reply:
Here is the response using the above prompt. 
![image](https://github.com/user-attachments/assets/2eb77a17-69b1-4384-b950-ab8d4b04b677)

Reply:
I am about to finish the PR and this one is blocking. Let me know your thoughts about it. Do you think we should modify the prompt for json? If the old one without json is working fine, and there is no issue, then this one isn't offering any added benefits.  But if you have noticed any incorrect mappings of relevancy value, then json can help. @0x4007

Reply:
I went ahead and implemented this. Specifications and comments are sent in a single json, and a response is returned from openai as json with key as comment id, and relevance as value. No more new line as delimiter for openai prompt.